### ISSUE TITLE
Discourse Staging Server for Update and Plugin Testing

### ISSUE THREAD

**Issue by kimardenmiller** [gomomentum/momentum_api#13]
Getting close to testing our new membership plugin and am thinking it would be a good time to consider having [a Staging Server as discussed here](https://discourse.gomomentum.org/t/discourse-maintenance-updates/5253/16?u=kim_miller).

I'm willing to do everything on our live server, but want to be sure that's the choice we are all making, not just me in the event something in a plugin brings down our live Discourse instance.

Thoughts @moe52 @ryan-hyer @sjmscott ?

**Comment by sjmscott** [comment]
I defer to Ryan and Moe.  I don’t understand all of the implications/risk.  I applaud you for creating a new membership plugin and I hope you can monetize it for yourself!

Steve

**Comment by kimardenmiller** [comment]
> I don’t understand all of the implications/risk.

I'd say the risk can be visualized as about what we experienced over a year ago when we needed Jay to bail us out of an upgrade gone wrong. While we may be more self-capable today to fix such an issue ourselves, a half day or 24 hours down would be the kind of thing such a staging server would help us avoid. 

We are adding into the mix code we have authored, but the risks are similar. Everyone in the D community talks about the fact that plugins can break your site. Without a staging server, we're likely to hit a future version of D that mixes with a plugin (ours or someone else's) and breaks the site.
